hathor/transaction/scripts/hathor_script.py

```diff
@@ -49,3 +58,30 @@ def pushData(self, data: Union[int, bytes]) -> None:
             self.data += (bytes([len(data)]) + data)
         else:
             self.data += (bytes([Opcode.OP_PUSHDATA1]) + bytes([len(data)]) + data)
+
+    def push_sighash(self, sighash: SighashType) -> None:
+        """Push a custom sighash to the script."""
+        match sighash:
+            case SighashAll():
+                pass
+            case SighashBitmask():
+                self.pushData(sighash.inputs)
+                self.pushData(sighash.outputs)
+                self.addOpcode(Opcode.OP_SIGHASH_BITMASK)
+            case SighashRange():
+                self.pushData(sighash.input_start)
+                self.pushData(sighash.input_end)
+                self.pushData(sighash.output_start)
+                self.pushData(sighash.output_end)
+                self.addOpcode(Opcode.OP_SIGHASH_RANGE)
+            case _:
+                assert_never(sighash)
+
+    def push_inputs_outputs_limit(self, limit: InputsOutputsLimit | None) -> None:
```

Comment:
It seems weird to accept None and simply do nothing. I guess the caller should take care of skipping this call.